src/Runtime/Distributions/Automata/ListAutomaton.cs

```diff
@@ -62,6 +88,18 @@ public class ListAutomaton<TElement, TElementDistribution>
         where TElementDistribution : class, IDistribution<TElement>, SettableToProduct<TElementDistribution>, SettableToWeightedSumExact<TElementDistribution>,
         CanGetLogAverageOf<TElementDistribution>, SettableToPartialUniform<TElementDistribution>, new()
     {
+        public ListAutomaton()
```

Comment:
Does this need to be public?

Reply:
Yes. Otherwise things won't compile. It was implicitly public before this change. But I had to add special constructor for use with ISerializable, and it shadows auto-generated constructors.